cmd/cli/util_test.go

```diff
@@ -157,6 +163,99 @@ var _ = Describe("Test getting pretty printed output of smi info of a list of me
 	})
 })
 
+func TestGetNamespaceOsmControllerPods(t *testing.T) {
+	testNamespace := "osm-namespace"
+
+	tests := []struct {
+		testName string
+		pods     []*corev1.Pod
+		podNames []string
+	}{
+		{
+			testName: "multiple pods (osm-controller pods and other pods) in multiple namespaces",
+			pods: []*corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osm-controller-pod-1",
+						Namespace: testNamespace,
+						Labels: map[string]string{
+							"app": constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osm-controller-pod-2",
+						Namespace: testNamespace,
+						Labels: map[string]string{
+							"app": constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-other-osm-controller-pod",
+						Namespace: "some-other-namespace",
+						Labels: map[string]string{
+							"app": constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-other-application-pod-1",
+						Namespace: testNamespace,
+						Labels: map[string]string{
+							"app": "myapp",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-other-application-pod-2",
+						Namespace: "some-other-namespace",
+						Labels: map[string]string{
+							"app": "myapp",
+						},
+					},
+				},
+			},
+			podNames: []string{
+				"osm-controller-pod-1",
+				"osm-controller-pod-2",
+			},
+		},
+		{
+			testName: "no pods",
+			pods:     []*corev1.Pod{},
+			podNames: []string{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			assert := tassert.New(t)
+
+			objs := make([]runtime.Object, len(test.pods))
+			for i := range test.pods {
+				objs[i] = test.pods[i]
+			}
+
+			fakeClientSet := fake.NewSimpleClientset(objs...)
+			podList := GetOSMControllerPods(fakeClientSet, testNamespace)
+
+			assert.Equal(len(test.podNames), len(podList.Items))
+			actualPodNames := make([]string, len(podList.Items))
+			for i, pod := range podList.Items {
+				actualPodNames[i] = pod.Name
+			}
+			sort.Strings(test.podNames)
+			sort.Strings(actualPodNames)
+			assert.Equal(test.podNames, actualPodNames)
```

Comment:
I think tassert may have a method for this, try assert.ElementsMatch? https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch